tsinfer/inference.py

```diff
@@ -1924,6 +1912,156 @@ def get_mutations(self, node_id):
+    then splitting up the grand MRCA (if keep_single_mrca is not ``True``), and finally
+    reordering the sample nodes such that they are the first nodes listed in the
+    node table,  removing tree nodes and edges that are not on a path between the
+    root and any of the samples (by applying the :meth:`tskit.TreeSequence.simplify`
+    method with ``keep_unary`` set to True but ``filter_sites``, ``filter_populations``
+    and ``filter_individuals`` set to False).
+
+    :param list[int] sample_order: A list of length ``ts.num_samples``
+        containing the ids of all the sample nodes in `ts` (e.g. as provided by the
+        :meth:`~tskit.TreeSequence.samples` method). The returned tree sequence will
+        contain the samples in this order. If ``None``, use ``ts.samples()``.
+    :param bool keep_single_mrca: If the oldest node is only a parent to a single
+        "grand MRCA" of all the samples, after removing the oldest node we by default
+        split the grand MRCA into separate nodes whenever its children change.
+        The rationale is that a grand MRCA consisting of a single ancestral haplotype
+        with all inference sites in the ancestral state is unlikely to represent a
+        single ancestor in the past. If this parameter is ``True``, any such grand
+        MRCA is *not* split (warning: this may cause problems when further dating the
+        inferred tree sequence).
+    """
+    if sample_order is None:
+        sample_order = ts.samples()
+
+    tables = ts.dump_tables()
+    if stabilise_node_ordering:
+        # Ensure all the node times are distinct so that they will have
+        # stable IDs after simplifying. This could possibly also be done
+        # by reversing the IDs within a time slice. This is used for comparing
+        # tree sequences produced by perfect inference.
+        times = tables.nodes.time
+        for t in range(1, int(times[0])):
+            index = np.where(times == t)[0]
+            k = index.shape[0]
+            times[index] += np.arange(k)[::-1] / k
+        tables.nodes.time = times
+
+    has_virtual_like_root = detach_node_over_grand_root(tables)
+    if has_virtual_like_root and not keep_single_mrca:
+        split_genomewide_mrca(tables)
+
+    logger.info(
+        "Sorting and simplifying with filter_sites=False, filter_populations=False, "
+        "filter_individuals=False, and keep_unary=True on "
+        f"{tables.nodes.num_rows} nodes and {ts.num_edges} edges"
+    )
+    tables.sort()
+    tables.simplify(
+        samples=sample_order,
+        filter_sites=False,
+        filter_populations=False,
+        filter_individuals=False,
+        keep_unary=True,
+    )
+    logger.info(
+        "Finished simplify; now have {} nodes and {} edges".format(
+            tables.nodes.num_rows, tables.edges.num_rows
+        )
+    )
+    return tables.tree_sequence()
+
+
+def detach_node_over_grand_root(tables):
```

Comment:
Split this into a function ``has_single_edge_over_grand_root`` . The actual taking off is trivial and can be done like:
```
if has_single_edge_over_grand_root(tables):
    tables.edges.truncate(tables.edges.num_rows - 1)
```

Reply:
Done, although we need to move the mutations afterwards too, of course